internal/reader/home_chain.go

```diff
@@ -114,24 +123,46 @@ func (r *homeChainPoller) poll() {
 }
 
 func (r *homeChainPoller) fetchAndSetConfigs(ctx context.Context) error {
-	var chainConfigInfos []ChainConfigInfo
-	err := r.homeChainReader.GetLatestValue(
-		ctx,
-		consts.ContractNameCCIPConfig,
-		consts.MethodNameGetAllChainConfigs,
-		primitives.Unconfirmed,
-		nil,
-		&chainConfigInfos,
-	)
-	if err != nil {
-		return err
+	var allChainConfigInfos []ChainConfigInfo
+	pageIndex := uint64(0)
+
+	for pageIndex < pageIndexHardLimit {
```

Comment:
I believe based on discussion we should not have a hard cap anymore. We can keep fetching all pages until there are no more configs left. Because our page size is 100, we will always be fetching at most 1 page